packages/vuetify/src/labs/VNumberInput/VNumberInput.tsx

```diff
@@ -75,13 +75,14 @@ export const VNumberInput = genericComponent<VNumberInputSlots>()({
     const model = computed({
       get: () => _model.value,
       set (val) {
-        if (val === null) {
-          _model.value = null
+        if (val === null || (typeof val === 'string' && !val)) {
```

Comment:
Clearing number via backspace currently results in warning:

```
[Vue warn]: Invalid prop: type check failed for prop "modelValue". Expected Number with value 0, got String with value "". 
  at <VNumberInput ref="input" model-value="" rules= (3) [ƒ, ƒ, ƒ]  ... > 
  at <VContainer> 
  at <VApp> 
  at <Playground> 
  at <App>
```

`modelValue` here should be strictly enforced with type `number | null`, probably try to set `_model.value` to `null` when backspace to empty string.

Reply:
Hey @yuwu9145, thanks for the review!

As I described in the PR description - the useValidation composable doesn't perform validation when the input value changes to null. When resetting the input, it sets the value to an empty string. Instead of assigning 0, the model should be reset to match this behavior. Using null for resetting is not viable because validation won't be triggered. Therefore, the model value should remain as the empty string in such cases to ensure proper validation behavior.

I think it might be worth of refactoring to extend the model type with string. WDYT?

Reply:
I still believe model value should be set to `null` in number input.

As for the issue of `null` doesn't trigger validation, it's a separate issue that should be resolved in validation composable [demo](https://play.vuetifyjs.com/#eNqFkk1vnDAQhv/KhAuJFJZDc0Lsqh9STz01Ui8hBwOTXavGuPaYbpXsf+/YfCxtssoFhvE7j1+/5uE5cbbJPxmzGTwmRVISdkYJwl2lAcohs+hMr50cEBolnNtWSXfMhKe+SqATx+y3bOnA3Q93d1USp3jO7L6jQwLCI8GTRNXCIJRHoB60VwraHp1OWWDlfo82rMpWkOx1mZuZUlvIz8TPnoDdXKCybfoDjnl6H+kX0RNwyAImi5ixBVBYrzAc8SEWG4u/vLTYPlbJLFGiRsWKb8IRaNHheano+hZVFi2xIuB/hPqsaBQKK2qFY2NX5msX81GHrCYNHxslm5/MqT1Rr7+ELw74ns8vteEs/gk0kHjqAiKGc0/27fl1dGtOqM/Xz60yX/0d/OkaKw1FLR5NbwlafBJeETyPPjh2UcD1DWx3cD31AGK2xayJnSnnYjLF8qursXx5gfRrvGrpFl16O8+elmqJu4CU0NGsOd1MRYd06Nv1xqtk2ePKDx2k2yxA2MaIX285x/rOcJr+Pzq+TyHRJcPkdBt2iY/Hv5dWKig=)

Reply:
#20765

This PR should aim to make number input work consistent with text field, but model value type is kept with `number | null`

Reply:
It happens because of [this condition in `useValidation` composable](https://github.com/vuetifyjs/vuetify/blob/master/packages/vuetify/src/composables/validation.ts#L157), but it was added over two years ago and I assume it can affect many places if we change it.

Reply:
Yeah, it's not a concern in VNumberInput so should be carefully resolved/tested in #20765

Reply:
Then we can close this PR or just solve the first part of the issue regarding resetting the value?

Reply:
Just need to resolve the issue regarding resetting the value to null, and ignore validation.

I have pushed a commit, see if you had any comment

Reply:
@yuwu9145 it's fine for me, thanks for your participation and prompt reply! :)